packages/task/src/browser/quick-open-task.ts

```diff
@@ -234,77 +234,41 @@ export class QuickOpenTask implements QuickOpenModel, QuickOpenHandler {
     async configure(): Promise<void> {
         this.items = [];
         this.actionProvider = undefined;
-        const isMulti: boolean = this.workspaceService.isMultiRootWorkspaceOpened;
 
         const token: number = this.taskService.startUserAction();
 
         const configuredTasks = await this.taskService.getConfiguredTasks(token);
         const providedTasks = await this.taskService.getProvidedTasks(token);
-
-        // check if tasks.json exists. If not, display "Create tasks.json file from template"
-        // If tasks.json exists and empty, display 'Open tasks.json file'
-        let isFirstGroup = true;
         const { filteredConfiguredTasks, filteredProvidedTasks } = this.getFilteredTasks([], configuredTasks, providedTasks);
         const groupedTasks = this.getGroupedTasksByWorkspaceFolder([...filteredConfiguredTasks, ...filteredProvidedTasks]);
-        if (groupedTasks.has(TaskScope.Global.toString())) {
-            const configs = groupedTasks.get(TaskScope.Global.toString())!;
-            this.items.push(
-                ...configs.map(taskConfig => {
-                    const item = new TaskConfigureQuickOpenItem(
-                        token,
-                        taskConfig,
-                        this.taskService,
-                        this.taskNameResolver,
-                        this.workspaceService,
-                        isMulti,
-                        { showBorder: false }
-                    );
-                    item['taskDefinitionRegistry'] = this.taskDefinitionRegistry;
-                    return item;
-                })
-            );
-            isFirstGroup = false;
+
+        const scopes: TaskConfigurationScope[] = [TaskScope.Global];
+        if (this.workspaceService.opened) {
+            scopes.push(TaskScope.Workspace);
+            if (this.workspaceService.isMultiRootWorkspaceOpened) {
```

Comment:
We are adding root folders only for multRootWorkSpaces but it the project is not multi root, we miss the entries so provided tasks are not listed for not multi root projects.

Reply:
Yes, good point. I changed the configured task system to treat a single folder root as 'workspace' scope, but provided tasks still treat them as folder scope. Would you prefer that the provided task system treat a single root as workspace scope, or adding folder scope even for single-root workspaces?

Reply:
It looks like VSCode's approach is to favor the folder interpretation, so I'll modify the logic to add workspace scope only if the workspace is saved.

Reply:
The new version looks fine to me, I tried it and it works well !